### DISCUSSION TITLE
Set center for Map Interface

### DISCUSSION THREAD

**Discussion by vidhav** [directus/directus#11590] [Feature Requests]
How about a way to paste coordinates, in the map interface, to center the view.

Finding a location can be easier by for example searching in Googe Maps, than it is to pan and zoom the map "manually".

You may have to be explicit about the order of longitude and latitude, since the different map clients aren't always consistent.

**Comment by benhaynes** [comment]
Is this for setting the value, or just centering the map viewport?

Set the value manually:
![Screen Shot 2022-02-11 at 8 13 48 AM](https://user-images.githubusercontent.com/522079/153597695-328aa9b3-aab3-4262-b646-70465de6e518.png)

Find a location in the viewport:
![Screen Shot 2022-02-11 at 8 14 10 AM](https://user-images.githubusercontent.com/522079/153597759-665eae0b-f19b-469b-bd60-b9a6c8330aa0.png)

**Reply by vidhav** [reply]
Both would be great!

When I try to edit the raw value, I just get `[object Object]`, so I'm assuming the GeoJSON value isn't parsed. I also can't get it to accept GeoJSON input.

I have not seen the "Find location..." input. Is it depending on other settings?

**Reply by benhaynes** [reply]
Ahh, yeah, it might be that the GEOJSON is not parsed... not sure how complex that data is under the hood, but we should be able to get a more useful value in the raw modal.

The search might be limited to MapBox... which requires an API Key in Project Settings. A nice step up from the free OpenStreetMaps options included by default.

**Reply by vidhav** [reply]
It would be great if the raw modal could handle GeoJSON. I have mostly worked with Postgres/PostGIS and I believe the default format for a geometry is [WKT](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry). You may easily cast it to GeoJSON with `select my_geometry::json` to get it as a GeoJSON geometry. The geometry type should also be able to handle GeoJSON as a value directly.

Seems like an API key from Mapbox is required. It is apparently using MapboxGeocoder to handle search and Mapbox do make excellent services. However, there is a [Maplibre alternative](https://github.com/maplibre/maplibre-gl-geocoder). I would be interested in trying this one for custom search endpoints!

**Reply by vidhav** [reply]
It does not seem like the Geocoder can accept coordinates as an input out of the box. Mapbox has an [example](https://docs.mapbox.com/mapbox-gl-js/example/mapbox-gl-geocoder-accept-coordinates/) where they pass a "localGeocoder" - a function that can intercept the search request - to enable entering coordinates.

This might be a bit hacky... If we could enter a GeoJSON value in the raw modal, a simple Point geometry will take care of centering the map at a specific location.

**Reply by rijkvanzanten** [reply]
The API currently already uses GeoJSON for the input/output of geometry data, so the "raw value" of the field should also be in GeoJSON format

**Reply by vidhav** [reply]
I think I found out why: Geometry has "geometry" as its Field.type. If you look in [getJSType](https://github.com/directus/directus/blob/a4f31697f49dac8fe5e4bcda9979680541b0c571/app/src/utils/get-js-type.ts#L13), there is no action for geometry and it returns `undefined`. By resolving geometry as `object`, you can edit the GeoJSON object in the raw value modal.

**Reply by rijkvanzanten** [reply]
Great find! Want to open a PR to resolve that? 🙂

**Reply by vidhav** [reply]
Sure! I'll do it :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
